hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -1045,6 +1014,17 @@ private <T> boolean isKeyPresentInTable(String keyPrefix,
         // Case 1: We found an entry, but no cache entry.
         if (cacheValue == null) {
           // we found at least one key with this prefix.
+          // There is chance cache value flushed when
+          // we iterate through the table.
+          // Check in table whether it is deleted or still present.
+          if (table.getIfExist(kv.getKey()) == null) {
+            if (!keyIter.hasNext()) {
+              break;
+            }
+            kv = keyIter.next();
+            continue;
+          }
```

Comment:
Nit: we could avoid duplicating steps for `keyIter` by converting to `if-else`.  Something like this:

```java
        // Case 1: We found an entry, but no cache entry.
        if (cacheValue == null) {
          // we found at least one key with this prefix.
          // There is chance cache value flushed when
          // we iterate through the table.
          // Check in table whether it is deleted or still present.
          if (table.getIfExist(kv.getKey()) != null) {
            // Still in table and no entry in cache
            return true;
          }
        } else if (cacheValue.getCacheValue() != null) {
          // Case 2a:
          // We found a cache entry and cache value is not null.
          return true;
        }
```

Reply:
Thanks for the suggestion, Updated.